\documentclass[12pt]{article}
\usepackage{graphicx}
\usepackage{epstopdf}
\usepackage{caption}
\usepackage{ctable}
\usepackage{multirow}
\usepackage{hhline}
%\usepackage[utf8]{inputenc}
\usepackage{algorithm}
\usepackage{algorithmic}
\usepackage{setspace} 
%\usepackage{fullpage}
\usepackage[top=1.5in, bottom=1.5in, left=1in, right=1in]{geometry}
\epstopdfsetup{outdir=./}
\usepackage{titlesec}

\titleformat{\section}
  {\normalfont\fontsize{12}{15}\bfseries}{\thesection}{1em}{}
\begin{document}


 \renewcommand{\thefootnote}{\fnsymbol{footnote}} 

\begin{center}
  {\bf Improving genome assemblies using multi-platform sequence data}\\

  P\i nar Kavak\,$^{1,2,}$\footnote{to whom correspondence should be addressed. pinar.kavak@tubitak.gov.tr, calkan@cs.bilkent.edu.tr}, Bekir Erg\"{u}ner\,$^{1}$, Bayram Y\"{u}ksel\,$^{3}$ , Duran \"{U}stek\,$^{4}$, Mahmut \c{S}amil Sa\u{g}\i ro\u{g}lu\,$^1$, 
  Tunga G\"{u}ng\"{o}r\,$^{2}$, and 
  Can Alkan\,$^{5,*}$ \\ 

  {\scriptsize
  $^{1}$Advanced Genomics and Bioinformatics Research Group (\.{I}GBAM), B\.{I}LGEM, %The Scientific and Technological Research Council of Turkey (
  T\"{U}B\.{I}TAK%)
, 41470 Gebze, Kocaeli, Turkey, pinar.kavak@tubitak.gov.tr;
  $^{2}$Department of Computer Engineering, Bo\u{g}azi\c{c}i University, 34342 Bebek, \.{I}stanbul, Turkey\\
  $^{3}$Advanced Genomics and Bioinformatics Research Group (\.{I}GBAM), MAM, %The Scientific and Technological Research Council of Turkey (
  T\"{U}B\.{I}TAK%)
, 41470 Gebze, Kocaeli, Turkey\\
  $^{4}$ Department of Medical Genetics, \.{I}stanbul Medipol University, 34810 Beykoz, \.{I}stanbul, Turkey\\
  $^{5}$Department of Computer Engineering, Bilkent University, 06800 Bilkent, Ankara, Turkey, calkan@cs.bilkent.edu.tr
  }
\end{center}

\begin{abstract}
\textit{De novo} assembly  using short reads generated by next generation sequencing technologies is still an open problem. Although there are several assembly algorithms developed for data generated with different sequencing technologies, and some that can make use of hybrid data, the assemblies are still far from being perfect. There is still a need for computational approaches to improve draft assemblies.
Here we propose a new method to correct assembly mistakes when there are multiple types of data obtained using different sequencing technologies that have different strengths and biases.
We apply our method to Illumina, 454, and Ion Torrent data, and also compare our results with existing hybrid assemblers, Celera and Masurca.
%{\bf Contact:} \href{pinar.kavak@tubitak.gov.tr}, \href{calkan@cs.bilkent.edu.tr}
\end{abstract}

\onehalfspacing
\vspace*{-0.3cm}
\section{Introduction}
Since the introduction of high throughput next generation sequencing (NGS) technologies, traditional Sanger sequencing is being abandoned especially for large-scale sequencing projects.
Although cost effective for data production, NGS also imposes increased cost for data processing and computational burden. In addition, the data quality is in fact lower, with greater error rates, and short
read lengths for most platforms. One of the main algorithmic problems to analyze NGS data is the \textit{de novo} assembly: i.e. ``stitching'' billions of short DNA strings into a collection of larger sequencees, ideally the size of chromosomes. However, ``perfect'' assemblies with no gaps and no errors are still lacking due to many factors,  including the short read and fragment (paired-end) lengths, sequencing errors in basepair level, and the complex and repetitive nature of most genomes. Some of these problems in {\textit de novo} assembly can be ameliorated through using data generated using different sequencing platforms, where each technology has ``strengths'' that may be used to fix biases introduced by others.

%Overlap-layout-consensus (OLC) graph based assemblers\cite{celera:2000, sga:2012} work well on the long read assembly. De-bruijn graph based assemblers\cite{velvetZerbino:2008, spadesBankevich:2012} are designed for short reads. There are also hybrid assemblers\cite{cabogMiller:2008,masurcaZimin:2013} which use multiple read libraries. 
%Pre-processing and post-processing operations before and after the assembly takes an important role on the assembly quality.

In this work, we propose to  improve draft assemblies (i.e. produced using a single data source, and/or single algorithm) by incorporating data generated using different NGS technologies, and applying novel correction methods. To achieve better improvements, we exploit the advantages of both short but low-error 
and long but erroneous reads. 
We show that correcting the contigs built by assembling long reads through mapping short (and high quality) read contigs produce the best results, compared to the assemblies generated by algorithms that use hybrid data.
\vspace*{-0.3cm}
\section{Methods}
\label{meth}

We first cloned a bacterial artificial chromosome (BAC) from human chromosome 13. We then sequenced this BAC separately using Illumina,  Roche/454, and Ion-Torrent platforms. Illumina data is paired-end, where the others are single-end. The read lengths are 101bp for Illumina, 10bp-1Kbp for Rochhe/454,   and 5bp-201bp for Ion Torrent. We also obtained a ``gold standard'' reference assembly using 
template-based assembly with Mira~\cite{mira} with Roche/454, which is then corrected with the Illumina reads. Since Roche/454 and Ion Torrent platforms have similar sequencing biases (i.e. problematic homopolymers), we worked on two separate groups: Illumina \& 454 and Illumina \& Ion-Torrent, which gives us an opportunity to compare Roche/454 and Ion-Torrent.

%\ctable[
%      cap     = {},
%      % 
%      caption = {Properties of the data},
%      % 
%      label   = {tab:dataprop},
%      doinside = \normalsize,
%      width = 0.95\textwidth,
%      %
%      %
%      % pos = h!tb,
%      star
%]      {>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}
%X>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}}
%{
%      \tnote[]{}
%
%}
%{ \FL
%Technology & Length range & Mean length & Mean seq. qual & Paired \\ \ML
%Illumina & 101bp & 101bp & 38 & paired \\
%\addlinespace[1mm]
%Roche/454 & 40bp-1027bp & 650bp & 28 & single-end \\
%\addlinespace[1mm]
%Ion-Torrent & 5bp-201bp & 127bp & 24 & single-end \\
%\LL
%}
\textbf{Pre-processing:} We first discarded the reads that has low average quality value (phred score 17, i.e. $\geq$2\% error rate). Next, we removed the reads with high N-density (with $>$10\% of the read consisting of Ns). We then trimmed groups of bases that seem to be non-uniform according 
to sequence base content. We also inevitably applied each assembler's pre-processing operations.

\textbf{Assembly:} We used several assembly tools: Velvet\cite{velvetZerbino:2008}, a de Bruijn graph based assembler to assemble the short reads; and two different overlap-layout-consensus (OLC) assemblers: Celera~\cite{celera:2000}, and SGA~\cite{sga:2012} to assemble the long read data sets (Roche/454 and Ion Torrent) separately. Finally, we also used a de Bruijn based assembler, SPAdes\cite{spadesBankevich:2012} on the long read data. 
We then mapped all draft assemblies to the E. coli reference sequence to identify and discard E. coli contamination due to the cloning process. At the end,
 we obtained one short read, and three long read assemblies.

\textbf{Correction:} We mapped the contigs obtained with the short reads onto the contigs generated by assembling long reads using BLAST\cite{blast}. Since BLAST may report multiple mapping locations due to repeats, we accepted only the ``best'' map locations. Reasoning from the fact that the short reads show less sequencing errors, we opted for the sequence reported by the short read based contigs over the long read contigs assemblies when there are disagreements between the pair, and patched the ``less fragmented'' long read assemblies. We repeated this process for each of the three long read assembly data sets.

\textbf{Evaluation:} We mapped each of the final corrected assemblies onto the reference genome we constructed, calculated various statistics based on the comparisons, and estimated assembly qualities (Table~\ref{tab:resultsTable}). We also used  two hybrid assemblers, Celera-CABOG~\cite{cabogMiller:2008} and Masurca \cite{masurcaZimin:2013} on the same data to compare our correction methodology with those of hybrid assembly algorithms.

%\begin{algorithm}
%\caption{Assemble the query (short reads contig) and the subject (long reads contig) according to mapping information}
%\label{assembly}
%\begin{algorithmic} 
%\REQUIRE {mapping query and subject}
%\IF{the map does not start at the beginning of the subject}
%\STATE{add the unmapping beginning of the subject}
%\ENDIF
%\IF{the map does not start at the beginning of the query}
%\STATE {add the first part of the query to the result with lowercase letters}
%\ENDIF
%\STATE{add the mapping part of the query}
%\IF{the map does not end at the end of the query}
%\STATE{add the last part of the query to the result with lowercase letters}
%\ENDIF
%\IF{the map does not end at the end of the subject}
%\STATE{add the unmapping end of the subject}
%\ENDIF
%\end{algorithmic}
%\end{algorithm}


\ctable[star
	    center
      cap     = {},
      % 
      caption = {Results of assembly correction method on BAC data.},
      % 
      label   = {tab:resultsTable},
      doinside = \tiny,
      width = 1\textwidth,
      %maxwidth=19cm,
      %
      %
       pos = htb      
       ]
       {c>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}
         X>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}
         X>{\raggedright\arraybackslash}X>{\raggedright\arraybackslash}}
       {         
         \tnote[]{{\tiny Name: the name of the data group that constitute the assembly; \# of contigs: the number of contigs that belong to the resulting assembly; \# of Mapped Contigs: the number of contigs that successfully mapped onto the reference sequence; \# of Covered bases: the number of bases on the reference sequence that are covered by the assembly; Coverage: percentage of covered reference; Avg. identity: percentage of the correctly predicted reference bases; \# of Gaps: The number of gaps that cannot be covered on the reference genome; Size of Gaps: total number of bases on the gaps.}}
         \tnote[*]{{\tiny ``2'' represents the results of the second cycle of correction, ``3'' represents the third cycle.}}
       }
       {
         \FL
         Name & Length & \# of Contigs & \# of Mapped Contigs & \# of Covered bases & Coverage & Avg. Identity & \# of Gaps & Size of Gaps\ML
		 \textbf{\textit{Reference}} & \textit{176.843} & & & & & & & \ML
		 \addlinespace
		 \textbf{Velvet} & & & & & & & \NN
         Ill. Velvet & 197,040 & 455 & 437 & 175,172 & 0.99055 & 0.97523 & 39 & 1,671 \ML
         \textbf{Celera} & & & & & & & \NN       
         454 Celera & 908,008 & 735 & 735 & 172,563 & 0.97580 & 0.92599 & 18 & 4,280 \NN
         Ion Celera & 39,347 & 27 & 27 & 47,638 & 0.26938 & 0.96932 & 47 & 129,205 \ML
         \addlinespace
         \textbf{Corrected Celera} & & & & & & & \NN
         Ill-454 Celera & 4,945,785 & 895 & 270 & 176,368 & 0.99731 & 0.94370 & 5 & 475 \NN
         Ill-454 Celera^2\tmark[*] & 5,078,059 & 890 & 265 & 176,640 & 0.998852 & 0.944527 & 4 & 203 \NN
         Ill-454 Celera^3 & 5,086,627 & 890 & 265 & 176,640 & 0.998852 & 0.944560 & 4 & 203 \NN
         Ill-Ion Celera & 93,909 & 30 & 28 & 81,819 & 0.46267 & 0.96327 & 36 & 95,024 \NN
         Ill-Ion Celera^2 & 145,262 & 30 & 28 & 91,962 & 0.52002 & 0.97412 & 33 & 84,881 \NN
         Ill-Ion Celera^2 & 216,167 & 30 & 28 & 99,645 & 0.56347 & 0.98066 & 34 & 77,198 \ML
         \textbf{SGA} & & & & & & & \NN
         454 SGA & 62,909,254 & 108,095 & 101,514 & 176,546 & 0.99832 & 0.97439 & 1 & 297 \NN
         Ion SGA & 842,997 & 6,417 & 6,122 & 153,092 & 0.86569 & 0.99124 & 197 & 23.751 \ML	
         \addlinespace
         \textbf{Corrected SGA} & & & & & & & \NN
         Ill-454 SGA & 295,009 & 335 & 335 & 176,757 & 0.99951 & 0.96823 & 5 & 86 \NN
         Ill-454 SGA^2 & 279,034 & 305 & 305 & 176,757 & 0.99951 & 0.96769 & 5 & 86 \NN
         Ill-Ion SGA & 197,509 & 291 & 291 & 175,052 & 0.98987 & 0.97501 & 45 & 1,791 \NN
         Ill-Ion SGA^2 & 203,064 & 291 & 291 & 175,676 & 0.99340 & 0.97413 & 34 & 1,167 \NN
         Ill-Ion SGA^2 & 204,524 & 291 & 291 & 175,677 & 0.99341 & 0.97405 & 34 & 1,166 \ML
         \textbf{SPADES} & & & & & & & \NN
         454 SPADES & 12,307,761 & 49,824 & 49,691 & 176,843 & 1.0 & 0.98053 & 0 & 0 \NN
         Ion SPADES & 176,561 & 110 & 107 & 167,890 & 0.94937 & 0.92909 & 9 & 8,953 \ML	
         \addlinespace
         \textbf{Corrected SPADES} & & & & & & & \NN
         Ill-454 SPADES & 290,702 & 298 & 298 & 176,454 & 0.99780 & 0.96538 & 5 & 389 \NN
         Ill-454 SPADES^2 & 290,917 & 297 & 297 & 176,454 & 0.99780 & 0.96530 & 5 & 389 \NN
%         Ill-454 SPADES^3 & 291.653 & 297 & 297 & 176.454 & 0.99780 & 0.96527 & 5 & 389 \NN
         Ill-Ion SPADES & 198,665 & 52 & 52 & 171,977 & 0.97248 & 0.94215 & 4 & 4,866 \NN
         Ill-Ion SPADES^2 & 200,307 & 52 & 52 & 172,101 & 0.97319 & 0.94230 & 2 & 4,742 \ML
         \textbf{Masurca} & & & & & & & \NN
         Ill-454 Masurca & 380 & 1 & 0 & 0 & 0 & 0 & 0 & 0 \NN
         Ill-Ion Masurca & 2,640 & 8 & 8 & 1,952 & 0.01104 & 0.98223 & 9 & 174,891 \ML
 		\textbf{Celera-CABOG} & & & & & & & \NN
         Ill-454 Celera & 1,101,716 & 891 & 891 & 174,330 & 0.98579 & 0.92452 & 12 & 2,513 \NN
         Ill-Ion Celera & 0 & 0 & 0 & 0 & 0.0 & 0.0 & 0 & 0.0 \ML
         \LL
       }
\vspace*{-0.3cm}
\section{Results and Conclusion}
\label{res}
We present a summary of the results in Table \ref{tab:resultsTable}. Briefly, the Velvet assembly using only the Illumina reads showed better coverage (99\%) and high average identity  (97.5\%) rates compared to Celera assembly using Celera. Correcting the Celera assembly with our method improves both coverage and average identity rates, which are then further improved by reiteratively applying our method. We also observe that the hybrid assemblers used in this study did not produce better results with this data set.

%The coverage of 454 assembly increases up to 99.7\% and the average identity rate increases up to 94.4\% on the first correction cycle. The repetitive correction cycles increase the coverage and average identity rates. We see that correcting the long read assembly with the short read contigs works well with all kind of assemblers. Corrected SGA assembly has the highest coverage rate among all. Hybrid assemblers Masurca and Celera did not give very good assembly rates on this data.

%Assembling short and long reads separately with de Bruijn and OLC assemblers and correcting them give better results than assembling short and long reads together with a hybrid assembler.
 Here we presented a new method to improve draft assemblies by correcting high contiguity assemblies using high quality short read contigs. However, the need to develop new methods that exploit
 different data properties of different NGS technologies remains.
       
%\section*{Acknowledgement}
%\noindent{\bf Funding\textcolon}
\paragraph{Funding\textcolon}
\footnotesize
The project is supported by the Republic of Turkey Ministry of Development Infrastructure Grant (no: 2011K120020), B\.{I}LGEM \-- T\"{U}B\.{I}TAK (The Scientific and Technological Research Council of Turkey) grant (no: T439000), and a T\"{U}B\.{I}TAK grant to C.A.(112E135).\\

%\bibliographystyle{elsarticle-harv}
\bibliographystyle{abbrv}
%\bibliography{ref}
%\bibliographystyle{plain}
\singlespacing
\vspace*{-1.5cm}
\begin{thebibliography}{}
\vspace*{-0.2cm}
\begin{tiny}
\bibitem[1]{celera:2000} E.W.Myers \textit{et~al} (2000) A Whole-Genome Assembly of Drosophila, \textit{Science}, {\bf 287}:2196-2204.
\bibitem[2]{sga:2012} J.Simpson \textit{et~al} (2012) Efficient de novo assembly of large genomes using compressed data structures, \textit{Genome Research}, {\bf 22}:549-556.
\bibitem[3]{velvetZerbino:2008} D.Zerbino, E.Birney (2000) Velvet: Algorithms for de novo short read assembly using de Bruijn graphs, \textit{Genome Research}, {\bf 18(5)}:821-829.
\bibitem[4]{spadesBankevich:2012} A.Bankevich {\it et~al} (2012) SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing, \textit{Journal of Computational Biology}, {\bf 19(5)}:455-477.
\bibitem[5]{cabogMiller:2008} J.R.Miller \textit{et~al} (2008) Aggressive assembly of pyrosequencing reads with mates, \textit{Bioinformatics}, {\bf 24(24)}:2818-2824.
\bibitem[6]{masurcaZimin:2013} A.Zimin \textit{et~al} (2013) The MaSuRCA genome Assembler, \textit{Bioinformatics}, {\bf 29(21)}:2669-2677.
\bibitem[7]{mira} B.Chevreux {\it et~al} (1999) Genome sequence assembly using trace signals and additional sequence information, \textit{Computer Science and Biology:Proceedings of the German Conference on Bioinformatics (GCB)}, {\bf 99}:45-56.
\bibitem[8]{blast} S.Altschul {\it et~al} (1990) Basic local alignment search tool, \textit{Journal of Molecular Biology}, {\bf 215(3)}:403-410.

\end{tiny}
%\bibitem[3]{zerbino:2009} D.Zerbino (2009) Genome assembly and comparison using de Bruijn graphs, \textit{University of Cambridge}.
%\bibitem[3]{allpaths:2008} J.Butler \textit{et~al} (2008) ALLPATHS: De novo assembly of whole-genome shotgun microreads, {\it Genome Research}, {\bf 18}:810-820.
%\bibitem[4]{eulerPevzner:2008} M.J.Chaisson \textit{et~al} (2008) De novo fragment assembly with short mate-paired reads: Does the read length matter?, \textit{Genome Research}, {\bf 19(2)}:336-346.
%\bibitem[5]{sharcgs:2007} J.C.Dohm \textit{et~al} (2007) SHARCGS, a fast and highly accurate short-read assembly algorithm for de novo genomic sequencing, \textit{Genome Research}, {\bf 17(11)}:1697-1706.
%\bibitem[6]{hapsemblerDonmez:2011} N.Donmez, M.Brudno (2008) Hapsembler: An Assembler for Highly Polymorphic Genomes, \textit{Proceedings of the 15th Annual International Conference on Research in Computational Molecular Biology}, {\bf pages}:38-52.
%\bibitem[7]{vcake:2007} W.R.Jeckl \textit{et~al} (2007) Extending assembly of short DNA sequences to handle error, \textit{Bioinformatics}, {\bf 23(21)}:2942-2944.
%\bibitem[8]{soapdenovo:2009} R.Li \textit{et~al} (2009) De novo assembly of human genomes with massively parallel short read sequencing, \textit{Genome Research}, {\bf 20(2)}:265-272.
%\bibitem[12]{abyssSimpson:2009} J.Simpson, R.Durbin (2009) ABySS: A parallel assembler for short read sequence data, \textit{Genome Research}, {\bf 19(6)}:1117-1123.
%\bibitem[13]{ssake:2007} R.L.Warren \textit{et~al} (2007) Assembling millions of short DNA sequences using SSAKE, \textit{Bioinformatics}, {\bf 23(4)}:500-501.

\end{thebibliography}
\end{document}
